### ISSUE TITLE
How to replace Weighted modulated convolution used in StyleGAN2 with AdaIN

### ISSUE THREAD

**Issue by ammar-deep** [rosinality/stylegan2-pytorch#257]
First of all a big thanks for providing such a wonderful code that works for all datasets.

StyleGAN2 updated the AdaIN used in StyleGAN1 because of the droplet issue found in some generated images and therefore introduced this modulated convolution operation. In my particular case I dont want to use this operation and instead use the AdaIN for simple style injection. 
So, I am trying to modify your code as I need the other components of StyleGAN2. I think I need to modify this class for what I want https://github.com/rosinality/stylegan2-pytorch/blob/bb459e012489eb4629004d486f05f65fde554ba7/model.py#L169 
Instead of this class I should replace it with a class having a convolution, AdaIN, and activation function?

I would be really thankful if you can give some suggestions on this modification?

**Comment by rosinality** [comment]
I think you can replace ModulatedConv2d with Conv + AdaIN.

**Comment by PangziZhang523** [comment]
Will there be water drop artifacts when replacing ModulatedConv2d with Conv + AdaIN? 
I happened to encounter this problem recently. When I used the encoder-decoder to generate the feature map, an artifact was generated. Is there any way to eliminate this artifact?Thank you！

**Comment by rosinality** [comment]
@PangziZhang523 As the whole reason of dropping adain is preventing blob artifacts, I think it will be hard to prevent it with adain.
Maybe batch norm or some other normalizations could perform better.

**Comment by ammar-deep** [comment]
@rosinality Thank you for the prompt response. I am proceeding as you suggested and as I guessed.

@PangziZhang523 I ran into a similar problem while developing an autoencoder for performing image reconstruction using a simple l1 distance. I was generating images with batchnorm and noticed a red&black blob in the bottom right corner. I corrected it by substituting instance norm for batchnorm.
You can try the other way round as @rosinality suggested.

**Comment by PangziZhang523** [comment]
I tried BN and LN, but there are still water drop artifacts. My artifact is in the form of half purple and half white. Is there any other way? Thanks Reply.

**Comment by ammar-deep** [comment]
Have you tried IN? Are you using image reconstruction in your auto-encoder? What losses are you using?

**Comment by PangziZhang523** [comment]
I used IN at the beginning, and my task is to regress a feature map. I don’t want any water drop artifacts on the feature map.I don't know what to do now.

**Comment by ammar-deep** [comment]
I think then you need to try the modulated convolution introduced in StyleGAN2. As it was introduced only to fix this water drop (blob) issue.

**Comment by PangziZhang523** [comment]
I'm trying to replace traditional convolution and IN with modulation convolution. However, it is not known whether to replace all or only the decoder part.

**Comment by ammar-deep** [comment]
Only for the decoder part. Use your original encoder with the default convolution and normalization.
